### PR TITLE
MAT-8943 remove some featureflags

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-8943](https://jira.cms.gov/browse/MAT-8943)
(Optional) Related Tickets:
https://github.com/MeasureAuthoringTool/madie-root/pull/318
https://github.com/MeasureAuthoringTool/madie-cql-library/pull/237/
https://github.com/MeasureAuthoringTool/madie-measure/pull/1106
https://github.com/MeasureAuthoringTool/madie-editor/pull/508

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/Store/featureFlagStore.ts [hunk 1]
```diff
@@ -8,38 +8,26 @@ export interface FeatureFlags {
...
   Locking: boolean;
-  EnhancedTextFormatting: boolean;
   qiCore7: boolean;
   TransferMeasure: boolean;
   TransferLibrary: boolean;
   MeasureHistory: boolean;
 }
 const initialState: FeatureFlags = {
-  qiCoreElementsTab: false,
+  qiCoreElementsTab: true,
```
Comment (jkotanchik-SB) [inline src/Store/featureFlagStore.ts]: Should the default values be changed?

File: src/Store/featureFlagStore.ts [hunk 2]
```diff
@@ -8,38 +8,26 @@ export interface FeatureFlags {
...
   qdmHideJson: true,
-  enableQdmRepeatTransfer: false,
+  enableQdmRepeatTransfer: true,
```
Comment (jkotanchik-SB) [inline src/Store/featureFlagStore.ts]: Same here, should the default change?

File: src/util/useCheckCanEdit.ts [hunk 3]
```diff
@@ -1,19 +1,12 @@
...
+const useCheckUserCanEdit = (createdBy: string, acls: Array<Acl>): boolean => {
   const { getUserName } = useOktaTokens();
   const userName = getUserName();
 
-  if (!draft && !editTestsOnVersionedMeasures) {
-    return false;
-  }
-
+  // Always allow editing if user is owner or has shared access
+  // (EditTestsOnVersionedMeasures feature flag removed - always enabled)
```
Comment (jkotanchik-SB) [inline src/util/useCheckCanEdit.ts]: This comment probably won't mean much to us in a couple of months. Suggest removing as the code self-documents the rules for editing.

The "always allow" comment is a nice readable summary of the logic below.

File: src/util/useCheckCanEdit.ts [hunk 4]
```diff
@@ -1,19 +1,12 @@
 import useOktaTokens from "../hooks/useOktaTokens";
 import { Acl } from "@madie/madie-models/dist/Measure";
 
-const useCheckUserCanEdit = (
-  createdBy: string,
-  acls: Array<Acl>,
-  draft: boolean = true,
-  editTestsOnVersionedMeasures?: boolean
-): boolean => {
+const useCheckUserCanEdit = (createdBy: string, acls: Array<Acl>): boolean => {
```
Comment (jkotanchik-SB) [inline src/util/useCheckCanEdit.ts]: The test case uses a third argument not in the signature. Should both states be tested, be defaulted, or removed from the test?

File: src/Store/featureFlagStore.ts [hunk 5]
```diff
@@ -28,15 +22,9 @@ const initialState: FeatureFlags = {
...
   stu6TestCaseValidation: false,
-  QDMIncludeRAVValues: false,
-  QICoreIncludeRAVValues: false,
   MeasureSearch: false,
   LibrarySearch: false,
-  EditTestsOnVersionedMeasures: false,
-  OverlappingValueSets: false,
-  MinimizeAlerts: false,
   Locking: false,
-  EnhancedTextFormatting: false,
```
Comment (etan-sb) [inline src/Store/featureFlagStore.ts:39]: EnhancedTextFormatting needs to be added back because its feature flag will be removed in https://jira.cms.gov/browse/MAT-9144

File: src/Store/featureFlagStore.ts [hunk 6]
```diff
@@ -8,15 +8,9 @@ export interface FeatureFlags {
...
   stu6TestCaseValidation: boolean;
-  QDMIncludeRAVValues: boolean;
-  QICoreIncludeRAVValues: boolean;
   MeasureSearch: boolean;
   LibrarySearch: boolean;
-  EditTestsOnVersionedMeasures: boolean;
-  OverlappingValueSets: boolean;
-  MinimizeAlerts: boolean;
   Locking: boolean;
-  EnhancedTextFormatting: boolean;
```
Comment (etan-sb) [inline src/Store/featureFlagStore.ts:19]: EnhancedTextFormatting needs to be added back because its feature flag will be removed in https://jira.cms.gov/browse/MAT-9144

**Review by etan-sb — DISMISSED** [review]

**Review by etan-sb — APPROVED** [review]
